### PR TITLE
Upgrade to 0.14.0 alpha2

### LINKED ISSUE #225: Upgrade to CosmWasm 0.14
When released

### LINKED ISSUE #229: Upgrade to CosmWasm v0.14.0

### PR DESCRIPTION
**Pull request by maurolacy**
Closes #229 .

A couple of tests in multi-test are failing, but publishing anyway as they are interesting cases to look at (~~wrong~~ sender can now send bank messages in the multi-test context).

**Update**: This makes sense, because `BankMsg` doesn't have a `from_address` anymore. Will just modify / remove these tests.

Still needs some styling, and closely following `MIGRATING.md` guidelines, but this is compiling, passes linting AFAIK, and passes ~~almost~~ all tests.

### CODE REVIEW

**Comment by ethanfrey** [comment]
I did some cleanup to see if this passes CI now.

I wonder if we should rename `HandleMsg` to `ExecuteMsg` across the board as well? This will change all the schema file output as well.

File: contracts/cw20-base/src/allowances.rs [hunk 1]
```diff
@@ -176,7 +179,8 @@ pub fn handle_burn_from(
         Ok(meta)
     })?;
 
-    let res = HandleResponse {
+    let res = Response {
```
Comment (ethanfrey) [inline contracts/cw20-base/src/allowances.rs:182]: Would it make sense to extend `Response::default()`?

Or do something like:

```
let mut res = Response::default();
res.add_attribute("action", "burn_from");
res.add_attribute("from", owner);
res.add_attribute("by", deps.api.human_address(&spender_raw));
res.add_attribute("amount", amount);
Ok(res)
```

We could do this consistently if you like (at least on a few contracts as demo and make an issue to finish it).

NB: I would prefer the shorter (and more Ethereum like) `res.emit("from", owner)` but Simon didn't like that as it wasn't clear. Curious as to your opinion.

**Review by ethanfrey — COMMENTED** [review]
Looking good. If there are some particular tests you want me to look at closer, just add a comment there.

I am curious about renaming `HandleMsg` -> `ExecuteMsg` and anything else we need to change like that for consistency

File: contracts/cw20-base/src/allowances.rs (hunk 1 above)
Comment (maurolacy) [reply contracts/cw20-base/src/allowances.rs:182]: I prefer it as it is now, for clarity.

A `default()` and overwriting is discouraged for style, by the way. See https://rust-lang.github.io/rust-clippy/master/#field_reassign_with_default 
 
A default can still be useful, though now that addresses don't have defaults, a "full default" is not possible anymore.

**Comment by maurolacy** [comment]
> I did some cleanup to see if this passes CI now.
> 
> I wonder if we should rename `HandleMsg` to `ExecuteMsg` across the board as well? This will change all the schema file output as well.

I think the sensible thing is to rename it. Particularly when writing new contracts, it would be confusing to have to write a `HandleMsg` for `execute`.

Maybe we can leave this change for a future version, though, as not to introduce so many changes at once.

**Comment by ethanfrey** [comment]
> > I did some cleanup to see if this passes CI now.
> > I wonder if we should rename `HandleMsg` to `ExecuteMsg` across the board as well? This will change all the schema file output as well.
> 
> I think the sensible thing would be to rename it. Particularly when writing new contracts, it would be confusing to have to write a `HandleMsg` for `execute`.
> 
> Maybe we can leave this change for a future version, though, as not to introduce so many changes at once.

We can do that in another PR, but let's do this by the next v0.6.0 cosmwasm-plus release, so that is a stable point.

**Comment by maurolacy** [comment]
Only missing piece AFAIK, was migrating to the new entry point system. Do the old system still works? If not, we should open a follow-up issue and finish that.

**Comment by ethanfrey** [comment]
Yeah, they both work and there is a new issue for that one. It's a bit down the backlog https://github.com/CosmWasm/cosmwasm-plus/issues/230 But if you want to, you can pull that next (it shouldn't be a big deal)

**Comment by maurolacy** [comment]
Just for reference / documentation, addressed pending changes in #249 and #250.

**Pull request merged**